### PR TITLE
fsinternetradio tests: migrate vom Groovy to pure Java

### PR DESCRIPTION
**Pull request by maggu2810**
Fixes: #3732

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/.classpath [hunk 1]
```diff
@@ -1,10 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="src" output="target/test-classes" path="src/test/groovy"/>
+	<classpathentry kind="src" path="src/test/java"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry exported="true" kind="con" path="GROOVY_SUPPORT"/>
-	<classpathentry exported="true" kind="con" path="GROOVY_DSL_SUPPORT"/>
 	<classpathentry kind="con" path="org.eclipse.m2e.MAVEN2_CLASSPATH_CONTAINER"/>
```
Comment (sjka) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/.classpath]: I know, it was there before - but just stumbled across while reading: as the project (correctly) does not have the m2e nature, isn't this here old garbage?

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    String RADIO_BINDING_ID = "fsinternetradio"; // taken from the binding.xml file
+    String RADIO_THING_TYPE_ID = "radio"; // taken from the thing-types.xml file
+    String DEFAULT_RADIO_THING_UID = String.format("%s:%s:%s", RADIO_BINDING_ID, RADIO_THING_TYPE_ID,
+            DEFAULT_RADIO_SERIAL_NUMBER);
+
+    @Before
+    public void setUp() {
+        discoveryParticipant = getService(UpnpDiscoveryParticipant.class, FSInternetRadioDiscoveryParticipant.class);
+        assertThat("The FSInternetRadioDiscoveryParticipant service cannot be found", discoveryParticipant,
```
Comment (sjka) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java]: I really don't like these string messages. They are completely agains the goal of the matchers which aim at producing readable code. To me, this reads like "assert That The FSInternetRadioDiscoveryParticipant service cannot be found" - which of course is the exact opposite of what should be asserted. Therefore IMHO we should drop them (everywhere - won't mention it again). wdyt?

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    }
+
+    /**
+     * Verify valid DiscoveryResult with FSInterntRadio device without base URL.
+     *
+     * @throws ValidationException
+     */
+    @Test
+    public void validDiscoveryResultIfWithoutBaseUrl() throws ValidationException {
+
```
Comment (sjka) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java]: empty line

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+                result.getProperties().get(FSInternetRadioBindingConstants.PROPERTY_MANUFACTURER),
+                is(DEFAULT_RADIO_MANIFACTURER));
+        assertThat("A result with wrong model number was created for a FSInternetRadio device without base URL",
+                result.getProperties().get(FSInternetRadioBindingConstants.PROPERTY_MODEL),
+                is(DEFAULT_RADIO_MODEL_NUMBER));
+
+    }
+
+    private RemoteDevice createDefaultFSInternetRadioDevice(URL baseURL) throws ValidationException {
+
```
Comment (sjka) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java]: empty line

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioHandlerOSGiTest.java [hunk 5]
```diff
@@ -0,0 +1,969 @@
...
+
+    private void turnTheRadioOn(Thing radioThing) {
+        radioHandler.handleCommand(radioThing.getChannel(FSInternetRadioBindingConstants.CHANNEL_POWER).getUID(),
+                OnOffType.ON);
+
+        FrontierSiliconRadio radio;
+        try {
+            Field f = radioHandler.getClass().getDeclaredField("radio");
+            f.setAccessible(true);
+            radio = (FrontierSiliconRadio) f.get(radioHandler);
```
Comment (sjka) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioHandlerOSGiTest.java]: Okay here, of course - so this is more meant to be a mental note for us: maybe hassle-free private field access is worth a utility method in our base class?

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/RadioServiceMock.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.smarthome.binding.fsinternetradio.internal.radio.FrontierSiliconRadioConstants;
+
+/**
+ * Radio service mock.
+ *
+ * @author Markus Rathgeb - Migrated from Groovy to pure Java, made more robust
+ *
+ */
+public class RadioServiceMock extends HttpServlet {
```
Comment (sjka) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/RadioServiceMock.java]: Maybe `RadioServiceDummy` in order to better distinguish it from Mockito-Mocks? Technically the name is correct, of course...

**Comment by maggu2810** [comment]
Thanks for having a look at.
For the most comments I have to admit, that I first want to migrate the tests itself to pure Java.
I fixed the most visible stuff but go not into such detail. My intention has been, that the all improvement could be done later and this one should just drop Groovy.
I will comment inline.

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/.classpath (hunk 1 above)
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/.classpath]: Could be removed IMHO, too.
I will remove it.

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java (hunk 2 above)
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java]: I agree, I don't think we need to add a custom string to the assertions at all.
If the assertion fails on a test, we need to look at the code to fix it.
And there is already "expected behavior" that is reported to not match.

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioHandlerOSGiTest.java (hunk 5 above)
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioHandlerOSGiTest.java]: I tried to remove all reflection used by the Groovy tests.
The only solution for this one has been to change the access level for the radio member to "no modifier" or protected.
If give me a go to change the access level we could drop that reflection.

File: extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/RadioServiceMock.java (hunk 6 above)
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/RadioServiceMock.java]: Okay for me - I kept the old name, but will change it.

**Comment by maggu2810** [comment]
@sjka All comments has been addressed.

**Comment by sjka** [comment]
Nice, thanks!

**Pull request merged**